### PR TITLE
Support user metadata on activities and child/scheduled workflows

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Support user metadata on activities and child/scheduled workflows

closes https://github.com/temporalio/sdk-go/issues/1673

### CODE REVIEW

**Review by cretz — APPROVED** [review]
LGTM, but may have to wait for a 1.26 server release for https://github.com/temporalio/temporal/issues/6412. Not sure why activity metadata test is failing though.

**Comment by Quinn-With-Two-Ns** [comment]
@cretz  https://github.com/temporalio/temporal/issues/6412 is already in `1.25`, the unit tests ere failing because I forgot to update them

File: internal/internal_command_state_machine.go [hunk 1]
```diff
@@ -727,6 +731,7 @@ func (d *childWorkflowCommandStateMachine) getCommand() *commandpb.Command {
 		command := createNewCommand(enumspb.COMMAND_TYPE_START_CHILD_WORKFLOW_EXECUTION)
 		command.Attributes = &commandpb.Command_StartChildWorkflowExecutionCommandAttributes{StartChildWorkflowExecutionCommandAttributes: d.attributes}
 		command.UserMetadata = d.startMetadata
+		fmt.Println("command.UserMetadata", command.UserMetadata)
```
Comment (yuandrew) [inline internal/internal_command_state_machine.go:734]: want to remove before merging?

File: internal/workflow.go [hunk 2]
```diff
@@ -1083,7 +1096,9 @@ func (wc *workflowEnvironmentInterceptor) ExecuteChildWorkflow(ctx Context, chil
 	options.SearchAttributes = workflowOptionsFromCtx.SearchAttributes
 	options.TypedSearchAttributes = workflowOptionsFromCtx.TypedSearchAttributes
 	options.VersioningIntent = workflowOptionsFromCtx.VersioningIntent
-
+	options.StaticDetails = workflowOptionsFromCtx.StaticDetails
+	options.StaticSummary = workflowOptionsFromCtx.StaticSummary
+	fmt.Println("$$$$$ options", options)
```
Comment (yuandrew) [inline internal/workflow.go:1101]: this one too?

**Review by yuandrew — APPROVED** [review]

**Pull request merged**